### ISSUE TITLE
As member, I would like to see where everyone is on a map

### ISSUE THREAD

**Issue by ultrasaurus** [18F/midas#672]
We recently added locations with geo-locations, so now we should be able to display the community on a map.

From a development perspective, we could start by just creating a page at /map that displays a map of the world and where everyone is

From a design perspective, it would be great to have some mockups about how we might integrate this into the site

**Comment by ultrasaurus** [comment]
I had a thought that we could let people pin themselves on the map as part of some prompts to fill out their profile at the beginning...  might be a fun intro to the site.

Added new related issue for this:  https://github.com/18F/midas/issues/700

**Comment by sefk** [comment]
I'm still a bit hazy on the use case for this feature, @ultrasaurus. I can think of two places where someone would want to see location info.

1. Looking at a project, left pane shows avatars for "Requestor and Participants". That could be moved to the larger right pane and replaced with avatars on a map. 
- People without locations would float in the corner? That's an important case to handle well since initially nobody will have locations.
- Would want a visual indication of the requestor, maybe an avatar with a colored border?
- Would have to handle clustering well, as that's the typical case -- bunch of volunteers in Mexico City, bunch in Washington D.C., none anywhere else.

2. Looking at your own profile, see other people near you. On the profile page the right pane is already a bit tall. Clicking on a profile would take you to that person's page? If I'm making a new connection with someone 

In the meantime I'm going to do the table-stakes feature which is just a /map endpoint of all profiles and add location to the user model and profile page. That way we can see where *all* registered users are. But I'm not so sure that's useful in and of itself vs. these other use cases.

**Comment by sefk** [comment]
I can use some design input, @dhcole and @ultrasaurus and @brandtheisey. I need to get all the users with their locations (technically, their location tag contents) to put on the map. There are a couple of ways to do this.

1. I don't see an endpoint to get all users, but I could add one. It could return a giant list of all the users similar to what you get from the /api/user call. Probably too much data to be useful, but most reusable. Portion of "user" response JSON below.

2. An all users endpoint, but with only location info to make it more sane. 

3. Add a specific "location" api controller with very specialized queries. What a really want is the list of locations, each with their own lat/long, with the list of who's at that location.

Or I'm open to other suggestions. Thanks.


```
{
"username": "sefklon@gmail.com",
"name": "Sef Kloninger",
"title": "",
"bio": "Test admin\ ",
"photoId": 1,
"photoUrl": null,
"isAdmin": true,
"disabled": false,
"passwordAttempts": 0,
"id": 6,
"createdAt": "2015-03-28T17:57:51.000Z",
"updatedAt": "2015-04-14T00:18:17.000Z",
"location": {
"tagId": 47,
"id": 68,
"createdAt": "2015-04-14T00:18:17.000Z",
"tag": {
"type": "location",
"name": "Fresno, California",
"data": {
"lat": "36.74773",
"lon": "-119.77237",
"source": "geonames",
"sourceId": "5350937",
"dstOffset": "-7",
"gmtOffset": "-8",
```

**Comment by ultrasaurus** [comment]
Looks like that got cut off at the end.  Agree we need an all users call.

I think we should omit
```
"disabled": false,   // we just omit these
"passwordAttempts": 0,   // this just seems weird
```

also, not sure what these are in the location:
```
"tagId": 47,
"id": 68,
```

**Comment by ultrasaurus** [comment]
also on visual design, I was thinking that we might visualize the number of participants in a location represented by the radius of the "dot"
-- on the high end the radius could be proportional to the square root of the number of participants (so we would see a visual difference between 25 and 36, but not 25 to 26
-- on the low end, I want to see a noticeable visual difference between 1 and 2, but higher than that it matter less 

Basically I think people should be able to glance at the map and think:
- wow!  look at me, I'm that dot on the map!  (if they are solo in a region)
- OMG!  there is someone else whose close, maybe we could get together for lunch?
- Wow. There's lots of people in my area

It won't suprise anyone to see a large circle around DC, we should see if we can make sure that one has is rendered at a z-order so that nearby dots (VA, Baltimore, are visible above it)

**Comment by sefk** [comment]
Early PR submitted, will refine. Screen shot

![map-672](https://cloud.githubusercontent.com/assets/621565/7175932/7f3955ae-e3ce-11e4-84e0-5419c8f37deb.png)

This shows four users who's cities are Iceland, Nairobi, Sacramento, and Fresno (to show overlaps).

**Comment by sefk** [comment]
Better screenshot
![usermap-110m](https://cloud.githubusercontent.com/assets/621565/7338011/2c74fc1a-ebf3-11e4-85bf-af3c31d1a97a.png)

**Comment by dhcole** [comment]
Closed by #781
